File: moderate/minimum-coins/minimum.pl
open(INFILE, "<$ARGV[0]") or die("Cannot open file $ARGV[0] for reading: $!");
while(chomp(my $line = <INFILE>)) {
    $coins = 0;
    while($line) {
        if ($line > 4) { $coins++; $line -= 5;}
        elsif ($line > 2) { $coins++; $line -= 3;}
        else { $coins++; $line--;}
    }
    print "$coins\n";
}
close(INFILE);

File: moderate/minimum-coins/minimum.py
import sys
test_cases = open(sys.argv[1], 'r')
for test in test_cases:
    change = int(test.strip())
    coins = 0
    while change != 0:
        if change > 4:
            coins += 1
            change -= 5
        elif change > 2:
            coins += 1
            change -= 3
        else:
            coins += 1
            change -= 1
    print coins
test_cases.close()
